test/skip_with_error_test.cc

```diff
@@ -119,7 +119,7 @@ ADD_CASES("BM_error_during_running", {{"/1/threads:1", true, "error message"},
 
 void BM_error_during_running_ranged_for(benchmark::State& state) {
   assert(state.max_iterations > 3 && "test requires at least a few iterations");
-  int first_iter = true;
+  [[maybe_unused]] bool first_iter = true;
```

Comment:
this is a c++17 extension, no?

our bazel build should be setting c++11 but it passed CI. i'm a bit surprised.

Reply:
I think that compilers tend to ignore attributes which they do not recognize, at least this is what it seems to be happening. Locally I added some random attribute like [[asdfasdf]] and it compiled just fine, although clang-tidy threw an error. So it seems that compilers ignore those attributes but clang-tidy recognizes them. This [[maybe_unused]] is to prevent clang-tidy from throwing *clang-analyzer-deadcode.DeadStores*, but it can be also done by adding `(void)first_iter` after `first_iter=false`.

Reply:
i spotted this because my local build (osx clang bazel (which sets c++11)) failed on this line.  my linux bazel build doesn't fail (which is why the CI bots aren't failing).

this may be opening up the question: can we support c++17 yet? :D  @LebedevRI ?